x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/ClaimConnectorSyncJobAction.java

```diff
@@ -0,0 +1,160 @@
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.application.connector.Connector;
+import org.elasticsearch.xpack.application.connector.action.ConnectorUpdateActionResponse;
+import org.elasticsearch.xpack.application.connector.syncjob.ConnectorSyncJob;
+import org.elasticsearch.xpack.application.connector.syncjob.ConnectorSyncJobConstants;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class ClaimConnectorSyncJobAction {
+    public static final ParseField CONNECTOR_SYNC_JOB_ID_FIELD = new ParseField("connector_sync_job_id");
+    public static final String NAME = "indices:data/write/xpack/connector/sync_job/claim";
+    public static final ActionType<ConnectorUpdateActionResponse> INSTANCE = new ActionType<>(NAME);
+
+    private ClaimConnectorSyncJobAction() {/* no instances */}
+
+    public static class Request extends ConnectorSyncJobActionRequest implements ToXContentObject {
+
+        private final String connectorSyncJobId;
+        private final String workerHostname;
+        private final Object syncCursor;
+
+        public String getConnectorSyncJobId() {
+            return connectorSyncJobId;
+        }
+
+        public String getWorkerHostname() {
+            return workerHostname;
+        }
+
+        public Object getSyncCursor() {
+            return syncCursor;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.connectorSyncJobId = in.readString();
+            this.workerHostname = in.readString();
+            this.syncCursor = in.readGenericValue();
+        }
+
+        public Request(String connectorSyncJobId, String workerHostname, Object syncCursor) {
+            this.connectorSyncJobId = connectorSyncJobId;
+            this.workerHostname = workerHostname;
+            this.syncCursor = syncCursor;
+        }
+
+        private static final ConstructingObjectParser<Request, String> PARSER = new ConstructingObjectParser<>(
+            "claim_connector_sync_job",
+            false,
+            (args, connectorSyncJobId) -> {
+                String workerHostname = (String) args[0];
+                Object syncCursor = args[1];
+
+                boolean hasSyncCursor = syncCursor != null;
+
+                return new Request(connectorSyncJobId, workerHostname, hasSyncCursor ? syncCursor : null);
```

Comment:
I feel like `hasSyncCursor` is excessive

Your constructor's 3rd arg is now: `syncCursor != null ? syncCursor : null` . 

Interesting fact: We call it "buttery butter" (masło maślane) in Polish

Reply:
Looks like some copy paste error 😬